File: app/src/main/java/com/example/designpatterndemo/creational_pattern/factory_pattern/FactoryProvider.java
package com.example.designpatterndemo.creational_pattern.factory_pattern;

import com.example.designpatterndemo.creational_pattern.abstract_factory_pattern.Circle;
import com.example.designpatterndemo.creational_pattern.abstract_factory_pattern.Rectangle;
import com.example.designpatterndemo.creational_pattern.abstract_factory_pattern.Shape;
import com.example.designpatterndemo.creational_pattern.abstract_factory_pattern.Square;

public class FactoryProvider {

    public static Shape getShape(String type) {
        if (type == null) {
            return null;
        }

        if (type.equalsIgnoreCase("RECTANGLE")) {
            return new Rectangle();
        } else if (type.equalsIgnoreCase("CIRCLE")) {
            return new Circle();
        } else if (type.equalsIgnoreCase("SQUARE")) {
            return new Square();
        }
        return null;
    }
}

File: app/src/main/java/com/example/designpatterndemo/design_patterns/creational_pattern/abstract_factory_pattern/FactoryProvider.kt
package com.example.designpatterndemo.design_patterns.creational_pattern.abstract_factory_pattern

class FactoryProvider {
    companion object {
        @JvmStatic
        fun getFactory(factoryType: String?): AbstractFactory? {
            if (factoryType == null) {
                return null
            }
            if (factoryType.equals("SHAPE", ignoreCase = true)) {
                return ShapeFactory()
            } else if (factoryType.equals("COLOR", ignoreCase = true)) {
                return ColorFactory()
            }
            return null
        }
    }
}